### ISSUE TITLE
A malformed url causes 100% CPU usage.

### ISSUE THREAD

**Issue by IanCaunce** [gorhill/uBlock#1559]
A malformed URL inside an `img` or `script` tag will cause ublock origin to consume 100% CPU and the popup window will not fully render.

![image](https://cloud.githubusercontent.com/assets/10065879/14455996/86a17558-009a-11e6-9cb6-210ff17ae1a9.png)
Only restarting chrome will resolve the issue. 

The following snippet will trigger the bug.
```html
<!DOCTYPE html>
<html>
    <body>
        <script src="http://www.test.com../something/something.js"></script>
    </body>
 </html>
```

To recreate the issue:
1. Save the snippet as a .html file
2. View the file in a browser
3. Click the ublock origin icon

Browser Version: `Google Chrome 49.0.2623.110 (64-bit Linux)`
uBlock Version: `1.6.8`

Default Settings Used

Filter List:
```
uBlock filters‎
uBlock filters – Badware risks‎
uBlock filters – Experimental‎
uBlock filters – Privacy‎
uBlock filters – Unbreak‎
Adblock Warning Removal List‎
Anti-Adblock Killer | Reek‎
EasyList‎
Peter Lowe’s Ad server list‎
Basic tracking list by Disconnect‎
EasyPrivacy‎
Fanboy’s Enhanced Tracking List‎
Malvertising filter list by Disconnect‎
Malware Domain List‎
Malware domains‎
Malware domains (long-lived)‎
Malware filter list by Disconnect‎
Spam404‎
Anti-ThirdpartySocial
Fanboy’s Annoyance List‎
Fanboy’s Social Blocking List‎
```

**Comment by gorhill** [comment]
Out of curiosity: how did you stumble onto this issue?

**Comment by IanCaunce** [comment]
I was debugging thirdparty code when I noticed that uBlock was blocking some of its assets. I clicked on the popup to set a whitelist rule but it failed to render and my computer started to run slow. Checking htop I noticed that a google chrome thread was causing my CPU to run at 100%. Chrome wouldn't navigate to any web page as it was "Waiting for uBlock origin extension". It was then just a matter of elimination until I found the offending snippet of code.
